docs/content/doc/features/webhooks.en-us.md

```diff
@@ -104,3 +112,64 @@ X-Gitea-Event: push
   }
 }
 ```
+
+### Example
+
+This is an example of how to use webhooks to run a php script upon push requests to the repository.
+In your repository Settings, under Webhooks, Setup a Gitea webhook as follows:
+
+- Target URL: http://mydomain.com/webhook.php
+- HTTP Method: POST
+- POST Content Type: application/json
+- Secret: 123
+- Trigger On: Push Events
+- Active: Checked
+
+Now on your server create the php file webhook.php
+
+```
+<?php
+
+$secret_key = '123';
+
+// check for POST request
+if ($_SERVER['REQUEST_METHOD'] != 'POST') {
+    error_log('FAILED - not POST - '. $_SERVER['REQUEST_METHOD']);
+    exit();
+}
+
+// get content type
+$content_type = isset($_SERVER['CONTENT_TYPE']) ? strtolower(trim($_SERVER['CONTENT_TYPE'])) : '';
+
+if ($content_type != 'application/json') {
+    error_log('FAILED - not application/json - '. $content_type);
+    exit();
+}
+
+// get payload
+$payload = trim(file_get_contents("php://input"));
+
+if (empty($payload)) {
+    error_log('FAILED - no payload');
+    exit();
+}
+
+// convert json to array
+$decoded = json_decode($payload, true);
+
+// check for json decode errors
+if (json_last_error() !== JSON_ERROR_NONE) {
+    error_log('FAILED - json decode - '. json_last_error());
+    exit();
+}
+
+// check authorization
+if ($decoded['secret'] != $secret_key) {
```

Comment:
You shouldn't rely on this field existing, it is currently deprecated as we are migrating towards using the checksum header to validate hook content. Would you be able to update this?

Reply:
I didn't know and I actually just learnt about it from you. How does the new system work? Is there any doc about it yet?

Reply:
Yes, this https://togithub.com/gogs/gogs/issues/4233 and this https://github.com/go-gitea/gitea/issues/3901 should provide details you are looking for :)

Reply:
interesting indeed, let me check that make changes

Reply:
@techknowlogick I've updated the code snippet and tested it. It works. I hope that's what you had in mind.